ding_frontend.module

```diff
@@ -112,3 +112,24 @@ function ding_frontend_query_alter($query) {
     }
   }
 }
+
+/**
+ * Implements hook_boot().
+ *
+ * Clear page cache if maintainance status has changed.
+ */
+function ding_frontend_boot() {
+  $ding_maintenance_toggle = variable_get('ding_maintenance_toggle', 0);
```

Comment:
Variables should be prefixed with the name of the responsible module e.g. `ding_frontend_maintenance_toggle`.